designs/operator-monitoring-plugin.md

```diff
@@ -0,0 +1,225 @@
+---
+title: controller-observation-plugin
+authors:
+  - "@Kavinjsir"
+reviewers:
+  - "@varsha"
+  - "@camila"
+approvers:
+creation-date: 2022-06-06
+last-updated: 2022-06-06
+status: provisional
+see-also:
+  - ""
+replaces:
+  - ""
+superseded-by:
+  - ""
+---
+
+# a-grafana-dashboard-plugin-to-observe-controller-metrics
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+
+## Open Questions [optional]
+
+- 1. What does the ideal approach for the plugin to create the grafana manifests?
+- 2. Should we scaffold a dir with the config to install the grafana? See: https://grafana.com/docs/grafana/latest/installation/kubernetes/
+- 3. Where should the default directory be scaffold?
+  - grafana/
+  - grafana/dashboards
+  - grafana/config
+  - OR inside of the config dir another grafana dir with the k8s manifests to be applied on the cluster when the plugin is used
+- 4. How is the plugin backwards compatible for users who have prometheus scaffolded?
+  - Is it compatible only with specific versions of prometheus? Or do can we have an existing kb project and add this plugin to scaffold grafana manifests smoothly. Do we need any other changes to an old project structure.
+
+## Summary
+
+This EP will provide a way for the operator author to visualize the runtime metrics in order to observe the operator status.
```

Comment:
```suggestion
This proposal aims to provide an optional helper for the operator author to visualize the runtime metrics which are exported by default in any project that is based on [controller runtime]() and enable the metrics endpoint to observe the operator status using Grafana.
```

Reply:
Varsha's suggestion + a little nits

Reply:
Also, see the suggestion in the Motivation section where we try to write down the value that it brings.